Modules/CLI/Groups/Groups.cxx

```diff
@@ -10,41 +10,44 @@
  *  Ilwoo Lyu, ilwoolyu@cs.unc.edu
  *************************************************/
 
+// STD includes
 #include <cstdlib>
+#include <iostream>
 #include <vector>
 #include <string>
 
-#ifdef WIN32
-#include "dirent.h"
-#else
-#include <dirent.h>
-#endif
+// VTK includes
+#include <vtksys/Directory.hxx>
+#include <vtksys/SystemTools.hxx>
+
 #include "GroupsCLP.h"
 #include "GroupwiseRegistration.h"
-#include <iostream>
 
 bool getListFile(string path, vector<string> &list, const string &suffix)
 {
-
-    DIR *dir = opendir(path.c_str());
-    if (dir != NULL)
+  vtksys::Directory dir;
+  std::string errorMessage = "";
+  dir.Load(path.c_str(), &errorMessage);
+  if (errorMessage != "")
+  {
+    cerr << "Failed to list directory " << path << ": " << errorMessage << endl;
+    return false;
+  }
+  for (unsigned long fileNum=0; fileNum < dir.GetNumberOfFiles(); ++fileNum)
+  {
+    std::string filename = std::string(dir.GetFile(fileNum));
+    std::string filepath = path + "/" + filename;
+    if (vtksys::SystemTools::FileIsDirectory(filepath))
     {
-        while (dirent *entry = readdir(dir))
-        {
-            string filename = entry->d_name;
-            if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
-            {
-                list.push_back(path + "/" + filename);
-            }
-        }
-        closedir(dir);
-        sort(list.begin(), list.begin() + list.size());
-        return true;
-    }else{
-        
-        cerr<<"The directory does not exist! "<<path<<endl;
-        return false;
+      continue;
+    }
+    if(filename.find(suffix) != string::npos && filename.find_last_of(suffix) == filename.size() - 1)
```

Comment:
I don't think the logic here is actually correct (the original logic, which your change replicates).

`std::string::find_last_of` has the following semantics (from https://en.cppreference.com/w/cpp/string/basic_string/find_last_of)

Finds the last character equal to one of characters in the given character sequence.

The current use of this is `if(! getListFile(landmark.c_str(), landmarkList, "vtk"))` which makes it seem like we are looking for a `.vtk` file, but in reality a `.mov` file would incorrectly match.

I think this needs to change to use something like `std::string::rfind` instead.

Reply:
I have made this update.